packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -531,6 +550,7 @@ function getLargestRawDataEntryFromEntity(
  */
 export function shrinkRawData(
   data: UploadDataLookup[keyof UploadDataLookup][],
+  logger: IntegrationLogger,
```

Comment:
I don't necessarily disagree with including a Logger into this function for all the logging we're wanting to do, but we'd originally planned on trying to keep this more as a pure utility function that would just return the results of the shrink process.

Reply:
I think `shrinkRawData` is a fairly meaty function that could fail in cases that we are detecting here. In those cases, it is important that we quickly determine the cause of the issue by dumping size data surrounding the batch and its largest entity. I think it is reasonable to include the logger in this function (and let the subsequent functions `shrinkRawData` calls be purely utility) in light of this.

Reply:
In that case, we can simplify the return value of this function and just log the results at the end, since we'll have access to a logger in here now.